ReactiveCocoa/Swift/Property.swift

```diff
@@ -118,6 +118,37 @@ public final class MutableProperty<Value>: MutablePropertyType {
 		observer.sendNext(initialValue)
 	}
 
+	/// Atomically replaces the contents of the variable.
+	///
+	/// Returns the old value.
+	public func swap(newValue: Value) -> Value {
+		return modify { _ in newValue }
+	}
+
+	/// Atomically modifies the variable.
+	///
+	/// Returns the old value.
+	public func modify(@noescape action: (Value) throws -> Value) rethrows -> Value {
+		lock.lock()
+		defer { lock.unlock() }
+
+		let oldValue = _value
```

Comment:
I think it is fine to repeat outselves here. The setter doesn’t need to do any action on the previous value.